packages/rum-core/src/domain/assembly.spec.ts

```diff
@@ -49,94 +49,132 @@ describe('rum assembly', () => {
     setupBuilder.cleanup()
   })
 
-  describe('events', () => {
-    it('should allow modification on sensitive field', () => {
-      const { lifeCycle } = setupBuilder
-        .withConfiguration({
-          beforeSend: (event) => (event.view.url = 'modified'),
+  describe('beforeSend', () => {
+    describe('fields modification', () => {
+      it('should allow modification on sensitive field', () => {
+        const { lifeCycle } = setupBuilder
+          .withConfiguration({
+            beforeSend: (event) => (event.view.url = 'modified'),
+          })
+          .build()
+
+        lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, {
+          rawRumEvent: createRawRumEvent(RumEventType.LONG_TASK, { view: { url: '/path?foo=bar' } }),
+          startTime: 0 as RelativeTime,
         })
-        .build()
 
-      lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, {
-        rawRumEvent: createRawRumEvent(RumEventType.LONG_TASK, { view: { url: '/path?foo=bar' } }),
-        startTime: 0 as RelativeTime,
+        expect(serverRumEvents[0].view.url).toBe('modified')
       })
 
-      expect(serverRumEvents[0].view.url).toBe('modified')
-    })
+      it('should reject modification on non sensitive and non context field', () => {
+        const { lifeCycle } = setupBuilder
+          .withConfiguration({
+            beforeSend: (event: RumEvent) => ((event.view as any).id = 'modified'),
+          })
+          .build()
 
-    it('should reject modification on non sensitive field', () => {
-      const { lifeCycle } = setupBuilder
-        .withConfiguration({
-          beforeSend: (event: RumEvent) => ((event.view as any).id = 'modified'),
+        lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, {
+          rawRumEvent: createRawRumEvent(RumEventType.LONG_TASK, {
+            view: { id: 'aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee' },
+          }),
+          startTime: 0 as RelativeTime,
         })
-        .build()
 
-      lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, {
-        rawRumEvent: createRawRumEvent(RumEventType.LONG_TASK, {
-          view: { id: 'aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee' },
-        }),
-        startTime: 0 as RelativeTime,
+        expect(serverRumEvents[0].view.id).toBe('aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee')
       })
 
-      expect(serverRumEvents[0].view.id).toBe('aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee')
-    })
+      it('should allow modification on context field for events other than views', () => {
```

Comment:
We should also add test for modification of `event.context` directly.

notably setting it to null, which seems to crash the SDK currently 😕
```ts
beforeSend: (event) => {
  event.context = null
}